instructor/common.py

```diff
@@ -0,0 +1,604 @@
+        request.session['assignment_name'] = request.POST['name']
+        if request.POST['based_on']:
+            request.session['based_on'] = request.POST['based_on']
+
+        if 'continue' in request.POST:
+            return redirect('common_course_setup')
+
+    all_assignments = models.Assignment.objects.all()
+    return render_to_response('instructor/assignment_setup.html',
+                              {'assignments': all_assignments,
+                               'assignment_name': request.session['assignment_name'],
+                               'based_on': request.session['based_on'],
+                               'new': request.session['new']},
+                                context_instance=RequestContext(request))
+
+
+def course_setup(request):
+    CourseForm = modelform_factory(models.Course, fields=['code', 'name'])
+
+    if request.method == "POST":
+        if 'back' in request.POST:
+            return redirect("common_assignment_setup")
+        elif 'course_pk' in request.POST:
+            '''Adding to an existing course'''
+            course = models.Course.objects.get(pk=request.POST['course_pk'])
+        else:
+            '''Create a new course'''
+            user = User.objects.get(username=request.user)
+            form = CourseForm(request.POST)
+            course = form.save(commit=False)
+            course.owner_id = user.id
+            course.save()
+
+        '''Create a new assignment'''
+        assignment_name = request.session['assignment_name']
+        assignment_id = assignment_name + datetime.datetime.now().strftime("%I:%M%p on %B %d, %Y")
+
+        if request.session['based_on']:
+            a = models.Assignment(course=course,
+                              name=assignment_name,
+                              assignment_id=assignment_id,
+                              basedOn=request.session['based_on'])
+        else:
+            a = models.Assignment(course=course,
+                              name=assignment_name,
+                              assignment_id=assignment_id)
+        a.save()
+        request.session['assignment_id'] = a.id
+        request.session['new'] = False
+
+        if 'save' in request.POST:
+            return redirect("common_course_modify")
+        else:
+            '''Continue'''
+            return redirect('common_assignments_edit_strains')
+
+    else:
+        all_courses = models.Course.objects.all()
+        form = CourseForm()
+        return render_to_response('instructor/course_setup.html',
+                                  {'courses': all_courses,
+                                   'form': form,
+                                   'new': request.session['new']},
+                                  context_instance=RequestContext(request))
+
+
+def assignments_edit(request, assignment_pk):
+    assignment = models.Assignment.objects.get(pk=assignment_pk)
+    if assignment.basedOn is not None:
```

Comment:
I'm guessing this is to handle multipage forms which you do with https://docs.djangoproject.com/en/1.7/ref/contrib/formtools/form-wizard/ or at the very least, it will probably be a lot easier to keep all this state tracking in one class, and have simple methods on it, like:
```python
assignment_workflow = AssignmentWorkflow(request.session)
return assignment_workflow.edit()
```
or similar.  This may be too much at this point to do, but a suggestion for the future